lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -282,7 +283,7 @@ public function setCacheMode($cacheMode)
      * The returned SQL syntax depends on the connection driver that is used
      * by this query object at the time of this method call.
      *
-     * @return string SQL query
+     * @return list<string>|string SQL query
```

Comment:
For 3.0, we should try to replace this method with one that always returns an array of strings.

Reply:
Doctrine noob here: I assume a single string is the SQL query, probably with placeholders and such, but what is an array of strings? And why would we prefer this for Doctrine 3?